src/cli.ts

```diff
@@ -53,7 +59,9 @@ cli({
 		}
 
 		// Load REPL
-		process.argv.push(require.resolve('./repl'));
+		if (!argv.flags.test) {
+			process.argv.push(require.resolve('./repl'));
+		}
```

Comment:
I would prefer an implementation where tsx just passes down `--test` to Node.js. In other words, `src/test-runner/util.ts` should not be added.

The reason it's not currently working is because tsx starts the REPL here if no arguments are passed in.

Instead, this condition should might be better as "if no arguments **or flags** are passed in, start the REPL". More specifically, we should check how Node.js behaves with each flag: do any of them complement the REPL or block it?

Reply:
Yes, I started out with exactly that - just prevent the REPL and pass down `--test whatever` to node.

Currently only `argv._` from the `typeFlag` call are passed down to node, effectively removing all flags. This behavior would need to be adjusted for a more generic approach, no? Something like "no flags, no matching command -> use `process.argv.slice(2)` instead".

Plus, the native `node --test` will not pick up typescript files unless a glob is provided. Just wanted to improve convenience here with the `test-runner/util`

Reply:
I'll gladly implement either approach 👍🏼  Just thinking about my own projects I'd rather have `tsx --test` pick up all typescript files as well